server/src/internalClusterTest/java/org/opensearch/remotestore/ReplicaToPrimaryPromotionIT.java

```diff
@@ -0,0 +1,118 @@
+        for (int i = 0; i < numIter; i++) {
+            final int numOfDoc = scaledRandomIntBetween(0, 200);
+            logger.info("num of docs in iter {} {}", numOfDoc, i);
+            if (numOfDoc > 0) {
+                try (
+                    BackgroundIndexer indexer = new BackgroundIndexer(
+                        indexName,
+                        "_doc",
+                        client(),
+                        numOfDoc,
+                        RandomizedTest.scaledRandomIntBetween(2, 5),
+                        false,
+                        null
+                    )
+                ) {
+                    indexer.setUseAutoGeneratedIDs(true);
+                    indexer.start(numOfDoc);
+                    waitForIndexed(numOfDoc, indexer);
+                    numOfDocs += numOfDoc;
+                    indexer.stopAndAwaitStopped();
+                    if (random().nextBoolean()) {
+                        // 90% refresh + 10% flush
+                        if (random().nextInt(10) != 0) {
+                            refresh(indexName);
+                        } else {
+                            flush(indexName);
+                        }
+                    }
+                }
+            }
+        }
+
+        ensureGreen(indexName);
+
+        // sometimes test with a closed index
+        final IndexMetadata.State indexState = randomFrom(IndexMetadata.State.OPEN, IndexMetadata.State.CLOSE);
+        if (indexState == IndexMetadata.State.CLOSE) {
+            CloseIndexResponse closeIndexResponse = client().admin().indices().prepareClose(indexName).get();
+            assertThat("close index not acked - " + closeIndexResponse, closeIndexResponse.isAcknowledged(), equalTo(true));
+            ensureGreen(indexName);
+        }
+
+        // pick up a data node that contains a random primary shard
+        ClusterState state = client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
+        final int numShards = state.metadata().index(indexName).getNumberOfShards();
+        final ShardRouting primaryShard = state.routingTable().index(indexName).shard(randomIntBetween(0, numShards - 1)).primaryShard();
+        final DiscoveryNode randomNode = state.nodes().resolveNode(primaryShard.currentNodeId());
+
+        // stop the random data node, all remaining shards are promoted to primaries
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(randomNode.getName()));
+        ensureYellowAndNoInitializingShards(indexName);
+
+        state = client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
+        for (IndexShardRoutingTable shardRoutingTable : state.routingTable().index(indexName)) {
+            for (ShardRouting shardRouting : shardRoutingTable.activeShards()) {
+                assertThat(shardRouting + " should be promoted as a primary", shardRouting.primary(), is(true));
```

Comment:
nit - we can use `assertTrue(message, condition)`?

Reply:
no strong opinion on either of these . Keeping it as per other `ReplicaToPrimaryIT` only for now.